packages/backend/src/models/CommentModel/CommentModel.ts

```diff
@@ -19,6 +22,79 @@ export class CommentModel {
         this.database = deps.database;
     }
 
+    private static parseComments(
+        commentsWithUsers: (DbDashboardTileComments &
+            Pick<DbUser, 'first_name' | 'last_name'>)[],
+        userUuid?: string,
+        canUserRemoveAnyComment?: boolean,
+    ) {
+        const commentsPerDashboardTile: Record<string, Comment[]> = {}; // Stores comments grouped by their dashboard_tile_uuid
+        const allComments: Record<string, Comment> = {}; // Fast access lookup for parent comments
+        const orphanReplies: Record<string, Comment[]> = {}; // Stores orphan replies keyed by their intended parent's commentId
+
+        commentsWithUsers.forEach((comment) => {
+            const uuid = comment.dashboard_tile_uuid;
+            if (!commentsPerDashboardTile[uuid]) {
+                commentsPerDashboardTile[uuid] = [];
+            }
+
+            const structuredComment: Comment = {
+                commentId: comment.comment_id,
+                text: comment.text,
+                replyTo: comment.reply_to ?? undefined,
+                user: { name: `${comment.first_name} ${comment.last_name}` },
+                createdAt: comment.created_at,
+                resolved: comment.resolved,
+                replies: [],
+                canRemove:
```

Comment:
I think we just need to re-add the `textHtml` here

Reply:
that is on a separate branch... sorry I'm not doing the things in the right order...

Reply:
updated with html